lib/fluent/plugin/in_tail.rb

```diff
@@ -590,6 +594,8 @@ def detach_watcher(tw, ino, close_io = true)
         target_info = TargetInfo.new(tw.path, ino)
         @pf.unwatch(target_info)
       end
+
+      @tails_rotate_wait.delete(tw)
```

Comment:
I believe this is not the responsibility of `detach_wacher()`.
Logic inside `detach_watcher()` should be independent of `rotate_wait`.

Reply:
Done @ 341d8b89eb7314ac72ad663907a8cc05904c21ea